include/aws/common/common.h

```diff
@@ -172,6 +172,18 @@ void aws_wrapped_cf_allocator_destroy(CFAllocatorRef allocator);
 AWS_COMMON_API
 void *aws_mem_acquire(struct aws_allocator *allocator, size_t size);
 
+/**
+ * Allocates many chunks of bytes into a single block. Expects to be called with alternating void ** (dest), size_t
+ * (size). The first void ** will be set to the root of the allocation. Alignment is assumed to be sizeof(intmax_t).
```

Comment:
We shouldn't lock ourselves into intmax_t alignment, if we know the max alignment on our platform we should use that.

Reply:
I know, I don't like either. In C11 you can do [`_Alignof`](https://en.cppreference.com/w/c/language/_Alignof)`(`[`max_align_t`](https://en.cppreference.com/w/c/types/max_align_t)`)`, but in C99 this was the closest thing I could find. If you know of another way to get default alignment, I'm all ears :smile:

That said, once we actually support aligned allocations, it'll be trivial to add another version of this that accepts trios of `ptr, size, alignment` and individually align each object.

Reply:
I think we should just add an actual aws_mem_aquire_aligned function when we need it.